ci-operator/config/netobserv/network-observability-console-plugin/netobserv-network-observability-console-plugin-main.yaml

```
@@ -0,0 +1,133 @@
+base_images:
+  go-builder-base:
+    name: release
+    namespace: openshift
+    tag: golang-1.21
+  ubi:
+    name: ubi
+    namespace: ocp
+    tag: "9"
+  web-builder-base:
+    name: ubi-minimal
+    namespace: ocp
+    tag: "9"
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.21
+  use_build_cache: true
+images:
+- dockerfile_path: Dockerfile
+  from: ubi
+  inputs:
+    go-builder-base:
+      as:
+      - docker.io/library/golang:1.21
+    web-builder-base:
+      as:
+      - docker.io/library/node:16-alpine
+  to: network-observability-console-plugin
+- build_args:
+  - name: BUILDSCRIPT
+    value: :standalone
+  dockerfile_literal: |
+    ARG BUILDSCRIPT=
+    ARG LDFLAGS=
+    ARG TARGETPLATFORM=linux/amd64
+    ARG BUILDPLATFORM=linux/amd64
+    FROM --platform=$BUILDPLATFORM docker.io/library/node:16-alpine as web-builder
+    USER node
+
+    ARG BUILDSCRIPT
+    ARG TARGETPLATFORM
+    ARG TARGETARCH=amd64
+    WORKDIR /opt/app-root
+
+    COPY --chown=node web/package.json web/package.json
+    COPY --chown=node web/package-lock.json web/package-lock.json
+    WORKDIR /opt/app-root/web
+    RUN npm ci
+
+    WORKDIR /opt/app-root
+    COPY --chown=node web web
+    COPY mocks mocks
+
+    WORKDIR /opt/app-root/web
+    RUN npm run format-all
+    RUN npm run build$BUILDSCRIPT
+
+    FROM --platform=$BUILDPLATFORM docker.io/library/golang:1.21 as go-builder
+
+    ARG TARGETPLATFORM
+    ARG TARGETARCH=amd64
+    ARG LDFLAGS
+    WORKDIR /opt/app-root
+
+    COPY go.mod go.mod
+    COPY go.sum go.sum
+    COPY vendor/ vendor/
+    COPY .mk/ .mk/
+    COPY cmd/ cmd/
+    COPY pkg/ pkg/
+
+    RUN CGO_ENABLED=0 GOARCH=$TARGETARCH go build -ldflags "$LDFLAGS" -mod vendor -o plugin-backend cmd/plugin-backend.go
```

```suggestion
    RUN CGO_ENABLED=0 go build -ldflags "$LDFLAGS" -mod vendor -o plugin-backend cmd/plugin-backend.go
```

GOARCH is not needed.

Are you sure about CGO_ENABLED=0 ?

---

cc @jpinsonneau any thoughts?

---

all of these are for multi arch build. You don't need these here indeed but the final goal is to rely on the console plugin repository Dockerfiles

---

Hi @jpinsonneau,

Those settings are partly historical (the GOARCH var in particular). Others (TARGETPLATFORM and TARGETARCH) have some 'meaning' only when you do some specific cross-building strategies that we don't use: in particular, 2 stage docker builds where the first stage is always amd64 and you rely on the Golang cross-building capabilities. The final stage will be of the final arch, but still you need qemu capabilities that are not supported by RHEL.

These settings are what we have done upstream in the kubebuilder to enable multiarch with DockerCE.

However, podman (and builds) do not ever set the TARGETPLATFORM and TARGETARCH automatically.

So, the final effect is totally the opposite: if we run the build generated by this configuration on an arm64 node, the build will fail as `ARG TARGETARCH=amd64` will never be overridden automatically.

Not setting it will solve the problem for native multiarch builds, which is what we want here: whatever the architecture of the node where you run the build, it will work.